From b894bbd594015a5fbcd88d0d333480cb84adeac9 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Wed, 20 Nov 2019 13:55:10 -0600
Subject: [PATCH 13/34] ARM64: dts: Add MFG EEPROM Support

The Phoenix SOM has an AT24 EEPROM attached to I2C3 which
contains information programmed from the factory about the
device.

This patch only enables the EEPROM in the device tree.
In order to make the EEPROM available, set CONFIG_EEPROM_AT24
in the defconfig

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 arch/arm64/boot/dts/freescale/lpd-imx8mm-som.dtsi | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/lpd-imx8mm-som.dtsi b/arch/arm64/boot/dts/freescale/lpd-imx8mm-som.dtsi
index 2c9ff4bab0ad..485e3d6aa2a7 100644
--- a/arch/arm64/boot/dts/freescale/lpd-imx8mm-som.dtsi
+++ b/arch/arm64/boot/dts/freescale/lpd-imx8mm-som.dtsi
@@ -183,6 +183,13 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_i2c3>;
 	status = "okay";
+
+	eeprom@50 {
+		compatible = "atmel,24c64";
+		pagesize = <32>;
+		read-only;	/* Manufacturing EEPROM programmed at factory */
+		reg = <0x50>;
+	};
 };
 
 &iomuxc {
-- 
2.17.1

